### DISCUSSION TITLE
Unable to find instrumental of a song after extracting

### DISCUSSION THREAD

**Discussion by win98se** [alexpado/genshin-music-unpacker#17] [Q&A] [answered]
I'm using 4.5.0 as base and 4.6.0 update package to find instrumental of [a song](https://www.youtube.com/watch?v=qBPdYyy5bBI), using `--diff --prefix`.

However, among the files starting with `[C]`, there is no match.

**Comment by alexpado** [comment]
(Rewriting the whole thing, I need to think correctly before commenting)

At each update, Mihoyo moves a lot of music around in the folder. What was correct a while back isn't anymore, so `[U]` quite often contains new music from an update, as the file was already used for an old song in previous version.

I can't really think of a solution software-wise to better counter this and detect all the new songs. This would probably require analyzing every music one by one with an AI or something to compare the old song collection and the new one.

Plus, sometime I'm pretty sure they keep some songs in patches that sometime happens days after the initial launch of version, which obviously isn't present in the update package.

To talk about the music you linked, I don't think you'll be able to find the instrumental of it, but rather the one you actually hear in game (if not in a cutscene file, which is outside the scope of my extractor), as I don't think Mihoyo went through the fun to sync singers, music and animations, but I might be wrong on this one

**Reply by win98se** [reply]
According to [this page](https://genshin-impact.fandom.com/wiki/...It%27s_the_Iridescent_Arataki_Rockin%27_for_Life_Tour_de_Force_of_Awesomeness!), it seems most likely the voice and the instrumental is split (even both sounds singing the lyric `Of rosy skies` are also split).

Also unlike most cutscenes, that one is definitely not a video file, instead it's rendered in real time. This makes it much harder for me to find the music (whether voiced or instrumental) out. :/

In addition, I've also found someone posting [this](https://www.youtube.com/watch?v=hY86POKC84A) (I don't even know how they manage to extract it), but I don't want to use it because YouTube always compresses the quality to 128 kbps.

**Reply by alexpado** [reply]
I'm currently extracting all the musics and voice-over (no diff, takes around 30min to do voiceovers) files to check on that. I'll get back to you with filenames if I find something

**Reply by alexpado** [reply]
Welp, I'm stupid and had a major time loss, I was extracting everything from the CN version because I didn't changed my settings... Welp, I still learnt that `Streamed-.pck` files contains a lot of sounds but also a lot of musics from events, so that might be a clue. I'm currently extracting those on the right version this time

**Comment by alexpado** [comment, marked as answer]
It's `Music27_13` that you are searching for :)

**Reply by win98se** [reply]
Thank you! Finally found it.
Also I didn't expect the track's length is actually 2:03... I've listened to every track ranging from 1:40 to 2:00 before I seeked help.

**Reply by alexpado** [reply]
Yeah, that's what took me quite a lot of time too, in the end I did almost all of the game OST ranging from 3:00 to this one, pain
